### PR TITLE
Batch job kill

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java [hunk 1]
```diff
@@ -1027,6 +1021,75 @@ public TerminationData killJob(JobId jobId) {
         return terminateJob(jobId, JobStatus.KILLED);
     }
 
+    public TerminationData killJobs(List<JobId> jobIds) {
+        jobIds.forEach(jobId -> jlogger.info(jobId, "killing job"));
+        List<JobData> jobDatas = lockJobs(jobIds);
+        if (jobDatas == null) {
+            return null;// TODO emptyData(jobId);
```
Comment (MaelAudren) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java]: I think that code with todo should not be merged into master.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java [hunk 2]
```diff
@@ -1027,6 +1021,75 @@ public TerminationData killJob(JobId jobId) {
         return terminateJob(jobId, JobStatus.KILLED);
     }
 
+    public TerminationData killJobs(List<JobId> jobIds) {
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java]: Javadoc ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 3]
```diff
@@ -1137,6 +1143,93 @@ public void killJob(InternalJob job) {
...
+                                                                         " task.finishedTime = :finishedTime, " +
+                                                                         " task.executionDuration = task.finishedTime - task.startTime " +
+                                                                         " where task.jobData.id in :jobIds " +
+                                                                         " and task.taskStatus = org.ow2.proactive.scheduler.common.task.TaskStatus.RUNNING " +
+                                                                         " and task.startTime > 0 and task.executionDuration < 0 ")
+                                                            .setParameterList("jobIds", jobIds)
+                                                            .setParameter("finishedTime", System.currentTimeMillis())
+                                                            .executeUpdate();
+
+            logger.trace(String.format("Kill jobs (%s) and tasks: %d %d %d %d %d",
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java]: Do the logger provide you with string formatting API ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java (hunk 2 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java]: sure

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java (hunk 1 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java]: done

**Pull request merged**